packages/autorest.go/src/gocodemodel/gocodemodel.ts

```diff
@@ -465,6 +498,10 @@ export function isFormBodyParameter(param: Parameter): param is FormBodyParamete
   return (<FormBodyParameter>param).formDataName !== undefined;
 }
 
+export function isFormBodyCollectionParameter(param: Parameter): param is FormBodyCollectionParameter {
+  return (<FormBodyCollectionParameter>param).formDataName !== undefined && (<FormBodyCollectionParameter>param).collectionFormat !== undefined;
```

Comment:
TypeScript question: can we define the interface such that callers can cast instead of calling this helper?

Reply:
We could.  I imagine it would look something like how M4 types have a [`type`](https://github.com/Azure/autorest.go/blob/main/packages/autorest.go/src/transform/transform.ts#L264) field and we switch/case on them.  Do you think that would be easier to use?  Personally, I've really been happy with using type guards as the compiler automatically converts `param` to `FormBodyCollectionParameter` (or whatever type predicate is defined in the type guard) for me if the shape of the data matches.

Reply:
No, I don't suppose that would be easier. This pattern just strikes me as odd because I expect the type system to have a simpler built-in way to tell whether an object satisfies some interface, and this function makes the criteria the object's data, not its shape. But this perception is probably due to my unfamiliarity with TypeScript and its idioms.